tsdb/shard.go

```diff
@@ -472,6 +506,11 @@ func (s *Shard) SeriesKeys(opt influxql.IteratorOptions) (influxql.SeriesList, e
 		return []influxql.Series{{Aux: auxFields}}, nil
 	}
 
+	s.mu.RLock()
```

Comment:
Although I should add that "~4x slower" is going from 23ns to 92ns per `Lock()`/`Unlock()` pair, so the extra time may not be noticeable compared to the larger operations, but reducing allocations is a good thing, right?